outport/factory/outportFactory.go

```diff
@@ -0,0 +1,86 @@
+package factory
+
+import (
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/epochStart"
+	"github.com/ElrondNetwork/elrond-go/epochStart/notifier"
+	"github.com/ElrondNetwork/elrond-go/outport"
+	driversFactory "github.com/ElrondNetwork/elrond-go/outport/drivers/factory"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+)
+
+var log = logger.GetOrCreate("outport/factory")
+
+// ArgsOutportFactory is structure used to store all components that are needed to create a new instance of
+// DriverHandler
+type ArgsOutportFactory struct {
+	ArgsElasticDriver  *driversFactory.ArgsElasticDriverFactory
+	EpochStartNotifier sharding.EpochStartEventNotifier
+	NodesCoordinator   sharding.NodesCoordinator
+}
+
+// CreateOutport will create a new instance of OutportHandler
+func CreateOutport(args *ArgsOutportFactory) (outport.OutportHandler, error) {
+	outportHandler := outport.NewOutport()
+
+	shardID := args.ArgsElasticDriver.ShardCoordinator.SelfId()
+
+	if shardID == core.MetachainShardId {
+		handler := epochStartEventHandler(outportHandler.SaveValidatorsPubKeys, args.NodesCoordinator)
+		args.EpochStartNotifier.RegisterHandler(handler)
+	}
+
+	err := subscribeDrivers(outportHandler, args)
+	if err != nil {
+		return nil, err
+	}
+
+	return outportHandler, nil
+}
+
+func subscribeDrivers(outport outport.OutportHandler, args *ArgsOutportFactory) error {
+	err := createElasticDriverAndRegisterIfNeeded(outport, args.ArgsElasticDriver)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createElasticDriverAndRegisterIfNeeded(
```

Comment:
For simplicity, you can move this logic to `subscribeDrivers` (optional).

Reply:
I prefer to keep this logic inside this method because when more drivers will be added I will make another method like this (createSomethingSomethingDriverAndRegisterIfNeeded that will handle the creation of driver) if I move logic in subscribeDrivers method will grow for every driver that will be added.